### PR TITLE
Fix XMLHttpRequest issues

### PR DESCRIPTION
**Pull request by perkinslr**
### Identify the Bug or Feature request
#3236 
#3235 

### Description of the Change

Changes the decoding for lib:// targets to UTF-8
Changes `fetch` to an `async` function, and internally `await`s the body of the Request object.  
Filters out the pseudo-header `:Status` from the responseHeaders for the `fetch` API if the underlying response fails.

### Possible Drawbacks

Binary data returned from lib:// targets will be improperly decoded.  Binary data should be stored in asset:// instead.

### Documentation Notes
Sending `body:` argument via `fetch` API now works.
`lib://` URIs now return sensible output.

### Release Notes

Examples:

- Sending `body:` argument via `fetch` API now works.
- `lib://` URIs now return sensible output.

---
This change is [Reviewable](https://reviewable.io/reviews/rptools/maptool/3237)

### CODE REVIEW

File: src/main/java/net/rptools/maptool/client/ui/htmlframe/MTXMLHttpRequest.java [hunk 1]
```diff
@@ -56,7 +56,9 @@ public void _getResponseHeaders(JSObject jheaders) {
 
     responseHeaders.forEach(
         (key, value) -> {
-          jheaders.setMember(key, value);
+          if (!key.equals(":Status")) {
```
Comment (kwvanderlinde) [inline src/main/java/net/rptools/maptool/client/ui/htmlframe/MTXMLHttpRequest.java:59]: Since this isn't mentioned explicitly in the bug reports, I have a couple questions:
1. What problems does it cause for this pseudo-header to be present?
2. Are there other pseudo-headers that should also be filtered out to avoid similar problems?

Comment (perkinslr) [reply src/main/java/net/rptools/maptool/client/ui/htmlframe/MTXMLHttpRequest.java:59]: Oops!  That should have been included in the `fetch` bug report.  
`:Status` is an intentionally invalid HTTP header which lets the macro writer explicitly set the response code.  It is also used internally to get the status code from the request processing out to the wrapper layer.  It *could* use an optional, or similar data type, but exposing those to the macro writers would be a bit of a trick.  By specification, invalid HTTP headers *must not* be returned to the client, so it *must* be stripped out before the headers are returned.  If the response is wrapped in a javascript `Response` object (as the `fetch` API does), `Response`'s constructor will throw an exception on invalid headers.  Note this is different from undefined / extension headers (valid headers must not start with `:`).  

The problem is the header filtering happens when the response body is received.  If there is an exception thrown between when the header default values are set and the response body is received, the body-received function is not called, so the headers do not get filtered.  When `fetch` then tries to handle the failed request, *it* chokes on the invalid header.

So to sum up:

1. It breaks the `fetch` API
2. No.

That said, I *think* it might still have a problem *outside* the `fetch` API on internal server errors.  Probably need to refactor it slightly to always explicitly filter the headers *without* calling the body-received function.

Comment (perkinslr) [reply src/main/java/net/rptools/maptool/client/ui/htmlframe/MTXMLHttpRequest.java:59]: Hm, nope, I can't see *why* the `MTXMLHttpRequest::recv` doesn't get called before the headers get retrieved.  Especially given that the critical `onreadystatechange` call which triggers the javascript-side header processing happens *after* the headers should be handled.

**Review by kwvanderlinde — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/3237)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @perkinslr)

**Review by kwvanderlinde — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/rptools/maptool/3237)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @perkinslr)
